lib/posix/posix_internal.h

```diff
@@ -31,10 +31,10 @@ enum pthread_state {
 	PTHREAD_DETACHED = PTHREAD_CREATE_DETACHED,
 	/* The thread is running and joinable. */
 	PTHREAD_JOINABLE = PTHREAD_CREATE_JOINABLE,
+	/* A joinable thread exited and its return code is available. */
+	PTHREAD_EXITED,
```

Comment:
This is mainly for the mental model (if TERMINATED comes below EXITED, it's a bit more intuitive)